src/node_perf.cc

```diff
@@ -264,26 +265,17 @@ void LoopIdleTime(const FunctionCallbackInfo<Value>& args) {
 
 void UvMetricsInfo(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
   uv_metrics_t metrics;
-
   // uv_metrics_info always return 0
   CHECK_EQ(uv_metrics_info(env->event_loop(), &metrics), 0);
-
-  Local<Object> obj = Object::New(env->isolate());
-  obj->Set(env->context(),
-           env->loop_count(),
-           Integer::NewFromUnsigned(env->isolate(), metrics.loop_count))
-      .Check();
-  obj->Set(env->context(),
-           env->events(),
-           Integer::NewFromUnsigned(env->isolate(), metrics.events))
-      .Check();
-  obj->Set(env->context(),
-           env->events_waiting(),
-           Integer::NewFromUnsigned(env->isolate(), metrics.events_waiting))
-      .Check();
-
-  args.GetReturnValue().Set(obj);
+  Local<Value> data[] = {
+      Integer::New(isolate, metrics.loop_count),
+      Integer::New(isolate, metrics.events),
+      Integer::New(isolate, metrics.events_waiting),
+  };
+  Local<Array> arr = Array::New(env->isolate(), data, arraysize(data));
+  args.GetReturnValue().Set(arr);
```

Comment:
Maybe doing the same we do for `hrtime` will be even faster since we don't need to create the array, what do you think?

Reply:
Wouldn't doing the same as hrtime require a BindingData buffer creation? It also uses FastAPI which does not support Uint64 (and even casting to uint32 as hrtime, it doesn't support typed array). Am I missing something?

Reply:
BindingData, yes, refs:

- c++: https://github.com/nodejs/node/blob/4379dfb1fdcd6a862ae5c3c74757efdc5bc11b0d/src/node_process_methods.cc#L564-L578
- js: https://github.com/nodejs/node/blob/4379dfb1fdcd6a862ae5c3c74757efdc5bc11b0d/lib/internal/process/per_thread.js#L58-L61

But you can also use the bigint buffer:

- c++ https://github.com/nodejs/node/blob/4379dfb1fdcd6a862ae5c3c74757efdc5bc11b0d/src/node_process_methods.cc#L580-L588
- js: https://github.com/nodejs/node/blob/4379dfb1fdcd6a862ae5c3c74757efdc5bc11b0d/lib/internal/process/per_thread.js#L62-L64

Reply:
It is unfortunate that an array is returned only to be immediately dismissed after being converted to an object. What about return an object directly?

Not actual working code, but idea:

```C++
Local<Object> obj = Object::New(env->isolate());
obj->Set(env->context(), String::NewFromUtf8Literal(isolate(), "loopCount"), Integer::New(isolate, metrics.loop_count));
...
```

Reply:
WIth the approach of `hrtime`, we only need to create the array once (stored in the binding data), then we just need to call C++ to update the values and use them directly in JS, the fastest option available.

Reply:
Initially, I was using this approach but creating the whole `BindingData` class to `node_perf.cc` and implementing the serialization (as it inherits from SnapshotableObject) was too much IMO.

What about doing it in a follow-up PR? I can do it with you, so we can have a better trade-off for gains

Reply:
That's ok for me, the initial comment was not blocking anyway.